src/Pim/Bundle/EnrichBundle/Controller/ProductRestController.php

```
@@ -0,0 +1,350 @@
+
+        if (0 === $violations->count()) {
+            $this->productSaver->save($product);
+
+            return new JsonResponse($this->normalizer->normalize($product, 'internal_api'));
+        } else {
+            $errors = $this->transformViolations($violations, $product);
+
+            return new JsonResponse($errors, 400);
+        }
+    }
+
+    /**
+     * Remove an optional attribute form a product
+     *
+     * @param int $productId   The product id
+     * @param int $attributeId The attribute id
+     *
+     * @AclAncestor("pim_enrich_product_remove_attribute")
+     *
+     * @throws NotFoundHttpException     If product is not found or the user cannot see it
+     * @throws AccessDeniedHttpException If the user does not have right to edit the product
+     * @throws BadRequestHttpException   If the attribute is not removable
+     *
+     * @return JsonResponse
+     */
+    public function removeAttributeAction($productId, $attributeId)
+    {
+        $product = $this->findProductOr404($productId);
+        if ($this->objectFilter->filterObject($product, 'pim:internal_api:product:edit')) {
+            throw new AccessDeniedHttpException();
+        }
+
+        $attribute = $this->findAttributeOr404($attributeId);
+
+        if (!$product->isAttributeRemovable($attribute)) {
+            throw new BadRequestHttpException();
+        }
+
+        $this->productManager->removeAttributeFromProduct($product, $attribute);
+
+        return new JsonResponse();
+    }
+
+    /**
+     * Find a product by its id or return a 404 response
+     *
+     * @param string $id the product id
+     *
+     * @throws NotFoundHttpException
+     *
+     * @return ProductInterface
+     */
+    protected function findProductOr404($id)
+    {
+        $product = $this->productManager->find($id);
+        $product = $this->objectFilter->filterObject($product, 'pim:internal_api:product:view') ? null : $product;
```

If the product exists but the user can't see it we should throw a 401 error, not a 404.